packaging/tags.py

```diff
@@ -26,14 +24,39 @@
 _32_BIT_INTERPRETER = sys.maxsize <= 2 ** 32
 
 
-@attr.s(frozen=True, repr=False)
 class Tag(object):
-    interpreter = attr.ib(converter=str.lower)
-    abi = attr.ib(converter=str.lower)
-    platform = attr.ib(converter=str.lower)
+
+    __slots__ = ["_interpreter", "_abi", "_platform"]
```

Comment:
Why the slot usage? Chances are there will not be memory pressure from having millions of `Tag` instances.

Reply:
I've done it because of this comment: https://github.com/pypa/packaging/issues/178#issuecomment-520487651, but I can drop it if I've misinterpreted @pradyunsg's comment.

Reply:
> Why the slot usage?

I don't expect memory pressure to be an issue for us here, but the inability to assign to arbitrary attributes of the `Tag` object is nice IMO.

I don't care strongly about this.

Reply:
@pradyunsg I actually have the reverse reaction. :) Say I wanted to attach the path to the wheel file that a tag was associated with. Using slots means I have to store it off the Tag object versus letting me do it directly there.

**But**, having said all of this, if there's going to be `__hash__` support then I guess it needs to be as immutable as possible. So @pradyunsg wins. :)